sources/lib/Model/ContainerModelLayerPooler.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/*
+ * This file is part of the PommProject/PommBundle package.
+ *
+ * (c) 2014 - 2016 Grégoire HUBERT <hubert.greg@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PommProject\PommBundle\Model;
+
+use PommProject\Foundation\Client\ClientPooler;
+use PommProject\ModelManager\ModelLayer\ModelLayerPooler;
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+
+/**
+ * ModelLayerPooler
+ *
+ * Pooler for ModelLayer session client.
+ *
+ * @package   ModelManager
+ * @copyright 2014 - 2016 Grégoire HUBERT
+ * @author    Miha Vrhovnik
+ * @license   X11 {@link http://opensource.org/licenses/mit-license.php}
+ * @see       ClientPooler
+ */
+class ContainerModelLayerPooler extends ModelLayerPooler
+{
+    use ContainerAwareTrait;
+
+    private $serviceMap;
+
+    /**
+     * Add mapping between model and a service
+     *
+     * @param string $class
+     * @param string $serviceId
+     */
+    public function addModelToServiceMapping($class, $serviceId)
```

Comment:
As you allow to create our own pooler service, this method should defined in an interface because you use it in the `ModelPass` compiler and if I defined my own pooler I must have this method or the compiler will fail otherwise